libraries/botbuilder-core/src/turnContext.ts

```diff
@@ -457,13 +462,27 @@ export class TurnContext {
         });
 
         return this.emit(this._onSendActivities, output, () => {
-            return this.adapter.sendActivities(this, output)
-                .then((responses: ResourceResponse[]) => {
-                    // Set responded flag
-                    if (sentNonTraceActivity) { this.responded = true; }
-
-                    return responses;
+            if (this.activity.deliveryMode === 'buffered') {
+                // Append activities to buffer
+                const responses: ResourceResponse[] = [];
+                output.forEach((a) => {
+                    this.bufferedActivities.push(a);
+                    responses.push({ id: undefined });
```

Comment:
Are we just supporting sending and receiving message activities?

The current implementation doesn’t support operations that rely on Activity.id. 

What is the list of operations we do support or don’t support with buffered activity functionality?

Reply:
Does the parent store the activity.id on behalf of the skill, does it send them asynchronously in relation to the parent’s turn...? via an Event activity?